src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -104,6 +104,18 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
             }
         }
 
+        /// <inheritdoc />
+        public override async Task<ConnectNewHeadersResult> ConsensusTipChangedAsync()
```

Comment:
I don't really think this is needed, we anyway trigger a resync at the end of header presented